pepdbagent/utils.py

```diff
@@ -104,3 +105,53 @@ def convert_date_string_to_date(date_string: str) -> datetime.datetime:
     :return: datetime format
     """
     return datetime.datetime.strptime(date_string, "%Y/%m/%d") + datetime.timedelta(days=1)
+
+
+def order_samples(results: dict) -> List[dict]:
+    """
+    Order samples by their parent_guid
+
+    :param results: dictionary of samples. Structure: {
+                            "sample": sample_dict,
+                            "guid": sample.guid,
+                            "parent_guid": sample.parent_guid,
+                        }
+
+    :return: ordered list of samples
+    """
+    # Find the Root Node
+    # Create a lookup dictionary for nodes by their GUIDs
+    guid_lookup = {entry["guid"]: entry for entry in results.values()}
+
+    # Create a dictionary to map each GUID to its child GUID
+    parent_to_child = {
+        entry["parent_guid"]: entry["guid"]
+        for entry in results.values()
+        if entry["parent_guid"] is not None
+    }
```

Comment:
Could this ever result in multiple? Since its a `for-loop` and creating a list?

Reply:
??
it's creating dict from list results, it's impossible that we will have 2 parent_guid same